packages/jazz-tools/src/implementation/schema.ts

```diff
@@ -88,7 +88,8 @@ function optionalRef<C extends CoValueClass>(
 
 function ref<C extends CoValueClass>(
     arg: C | ((_raw: InstanceType<C>["_raw"]) => C),
-): co<InstanceType<C> | null>;
+    options?: never,
+): co<InstanceType<C>>;
```

Comment:
this means that co.ref fields on the class where they are used are always defined type-wise, right? This may be incorrect, because a value might not be loaded yet and return `null` when get-accessed.

So the type for the field has to be `InstanceType<C> | null` and while we might not be able to do anything about not *assigning* null, we can prevent it in the constructor (`CoMapInit`) at least?

Reply:
Oops, I missed that. Thanks for catching it!

Added an additional test to ensure the nullability of the ref properties and gone with the solution that you proposed.